packages/Modal/src/components/Header.js

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Button from "@paprika/button";
+import Heading from "@paprika/heading";
+import * as styled from "./Header.styles";
+
+const propTypes = {
+  children: PropTypes.node.isRequired,
+  hasCloseButton: PropTypes.bool,
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+  onClose: PropTypes.func,
+};
+
+const defaultProps = {
+  hasCloseButton: true,
+  level: 3,
+  onClose: () => {},
+};
+
+const Header = React.forwardRef((props, ref) => {
+  const { children, level, hasCloseButton, onClose, ...moreProps } = props;
+
+  return (
+    <styled.Wrapper ref={ref} {...moreProps}>
```

Comment:
Can we call it `<styled.Header>` instead?

Reply:
Say we use BEM methodology. Than we will have following structure: `.header > .header__wrapper` (not `.header > .header__header`).

When I think about semantic, it feels a bit strange for me having `Header` inside `Header`.

Reply:
I think this is minor, is ok going with wrapper mike :)

Reply:
👌